docs/tutorial/batch-minting.md

```diff
@@ -0,0 +1,268 @@
+related: 
+  'Lazy minting': https://nftschool.dev/tutorial/lazy-minting/
+---
+
+ # Batch-minting NFTs on Ethereum
+
+ This tutorial addresses some techniques in batch-minting big numbers of Ethereum non-fungible tokens (NFTs) as well as common idioms of uploading files and metadata on to IPFS/Filecoin via NFT.storage.
+
+See also: [Lazy minting](./lazy-minting.md)
+
+Quite often,  NFT projects consist of hundreds if not thousands of files and metadata starting their lives out on a computer file system, waiting nervously to be part of a mint.
+
+There are cases in which it is desirable to mint many NFTs all at once instead of doing each of them in a [just-in-time](./lazy-minting.md) fashion. Large single drops are often for more than just art, including scenarios like large PFP (profile picture) drops, event tickets, game packs, and etc.
+
+Some challenges that lie in minting a bulk of NFTs include high gas fees on the Ethereum network, synchronizing between mint and upload failures and recovering from them, and performance of uploading files to NFT.storage, etc.
+
+## Batch-uploading NFT assets
+
+Here is an overview of strategies to upload to NFT.storage in bulk.
+
+### Upload files iteratively
+
+The most straightforward, yet more involved way to upload files and metadata to NFT.storage is by iterating over all of them and calling [`NFTStorage.store`](https://nftstorage.github.io/nft.storage/client/classes/lib.NFTStorage.html#store) one-by-one. Being simple as it is, that means you are responsible for properly handling errors that might occur for each successive request and synchronize it with the minting process.
+
+```javascript
+
+async function readAndUpload() {
+  const promises = Array.from({length: 10}, (v, i) => i)
+      .map(async (i) => {
+        const image = (await fetch(`$creature-{i}.jpg`)).blob()
+        return {
+          image,
+          name: `$creature-{i}`,
+          description: `Metadata of creature-${i}`,
+        }
+      })
+
+  const client = new NFTStorage({ token })
+  const results = await Promise.all(ps)
+
+  await Promise.all(results.forEach((result) => {
+    const metadata = await client.store(nft)
+    console.log('NFT data stored!')
+    console.log('Metadata URI: ', metadata.url)
+  }))
+}
+
+```
+
+### Upload directory of files
+
+A simpler way is to use the [NFTStorage.storeDirectory](https://nftstorage.github.io/nft.storage/client/classes/lib.NFTStorage.html#storeDirectory) method to upload all the files stored in a local directory. This method handles many things under the hood for you, including rate-limiting for each request.
+
+```javascript
+
+import path from 'path'
+import { NFTStorage } from 'nft.storage'
+import { fileFromPath } from 'files-from-path'
+
+async function readAndUploadDir() {
+  const files = filesFromPath(dirpath, {
+		pathPrefix: path.resolve(dirpath),
+		hidden: true,
+	})
+  const storage = new NFTStorage({ token })
+  const cid = await storage.storeDirectory(files)
```

Comment:
Should we also be clear that using `storeDirectory` will only give you CID to that Folder. So developers still need to create metadata for their NFT collection and upload metadata again, right? 

Since `store` will take care of NFT assets as well as metadata, maybe we should give them more instructions if `storeDirectory` works different as `store`.

Reply:
From @alanshaw:

> Note that `store` is for uploading ERC-721 compatible metadata, linked to NFT assets whereas `storeDirectory` and `storeBlob` just store raw asset data (you need to create your own metadata and link it correctly to the asset).

Seems important to clarify in this guide.